geoapps/io/input_file.py

```diff
@@ -166,14 +167,19 @@ def write_ui_json(
             Writes default values stored in ui_dict to file.
         name: optional
             Name of the file
-        param_dict : optional
-            Parameters to insert in ui_dict values.
-            Defaults to the :obj:`InputFile.data` is not provided.
         workspace : optional
             Provide a workspace_geoh5 path to simulate auto-generated field in Geoscience ANALYST.
         """
 
         out = deepcopy(ui_dict)
+        if forward:
+            out["forward_only"] = True
+            inversion_type = out["inversion_type"]
+            if inversion_type == "mvi":
+                from geoapps.io.MVI.constants import forward_parameters as fparams
+            if inversion_type == "gravity":
+                from geoapps.io.Gravity.constants import forward_parameters as fparams
+            out = {k: out[k] for k in fparams}
```

Comment:
Brilliant. So we are basically just sub-sampling the full UI right?

Reply:
Exactly - doesn't fit the description of the issue, but I think it accomplishes what we needed with less code duplication than if we wrote a separate driver.